### PR TITLE
Update getAccessToken to support new error types

### PR DESCRIPTION
**Pull request by rleojoseph**
# Description
Updated `getAccessToken` method to return new error type `MASDKAccessTokenError`

`MASDKAccessTokenError` will support following errors,

* FailedToConformToProtocol
* AudienceNotSupportedError
* ScopesNotSupportedError
* AuthorizationFailureError
* Error

## Links
MINI-3132

# Checklist
- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `fastlane ci` without errors

### CODE REVIEW

File: MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift [hunk 1]
```diff
@@ -150,3 +150,54 @@ public enum MASDKCustomPermissionError: String, MiniAppErrorProtocol {
...
+        switch self {
+        case .failedToConformToProtocol:
+            return MASDKLocale.localize(.failedToConformToProtocol)
+        case .audienceNotSupportedError:
+            return "The value passed for 'audience' is not supported."
+        case .scopesNotSupportedError:
+            return "The value passed for 'scopes' is not supported."
+        case .authorizationFailureError(let description):
+            return description
+        case .error(let description):
```
Comment (Climbatize) [inline MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift:183]: in JS SDK we called this kind of errors "Other", do you think we can bring this to our own SDK?

File: MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift [hunk 2]
```diff
@@ -150,3 +150,54 @@ public enum MASDKCustomPermissionError: String, MiniAppErrorProtocol {
...
+            return description
+        }
+    }
+
+    /// Title of the error
+    public var name: String {
```
Comment (Climbatize) [inline MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift:189]: maybe we could rename this one `rawValue` to mimic a string enum behavior?

**Review by Climbatize — APPROVED** [review]
LGTM, just some trivial comments

File: MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift (hunk 1 above)
Comment (rleojoseph) [reply MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift:183]: Yeah, not a problem. But I sort of followed the confluence documentation. Should we update the documentation as well ?

File: MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift (hunk 2 above)
Comment (rleojoseph) [reply MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift:189]: No, these error is conformed to `MiniAppErrorProtocol` which has `name` & `description`. We can update them to `rawValue` but that involves in updating the `MiniAppErrorProtocol` itself. Is that okay ?

**Review by laubengaier — APPROVED** [review]
LGTM! 👍

File: MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift (hunk 2 above)
Comment (Climbatize) [reply MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift:189]: no that's fine, thanks for the heads up ;)

File: MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift (hunk 1 above)
Comment (Climbatize) [reply MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift:183]: I think that's not that important, as what I did on JS side is that if I receive an unknown error I set is as `Other` type. We can go with this implementation.

**Pull request merged**